tests/cmake_integration/Messenger/Messenger_1/CMakeLists.txt

```diff
@@ -36,19 +36,17 @@ foreach(t ${all_targets})
 endforeach()
 
 # Copy configs/scripts into build-output directory
-if (EXISTS ${DDS_ROOT}/tests/DCPS/Messenger)
```

Comment:
This if-block exists to enable these tests to work outside of the source tree (e.g. with a make-installed OpenDDS)—these should not be removed.

Reply:
If we're using this file which is in the tests directory, then why wouldn't we be able to access Messenger, even if OpenDDS itself is installed somewhere else?

Reply:
To elaborate on that, I changed it so it's copying from a location relative to this `CMakeLists.txt`. So unless this file is being copied somewhere else, I don't see how this won't work.

Reply:
For example, in the [Example Using Installed OpenDDS](https://github.com/objectcomputing/OpenDDS/blob/master/docs/cmake.md#example-using-installed-opendds-unix-only) for using CMake outside of the source tree, a make-installed version does not have anything under ${DDS_ROOT}/tests, which is what the target created within that if-block depends upon.

Reply:
Alright, yes, I would be breaking that particular example. However, why copy `tests/cmake_integration/Messenger` somewhere else and not just do it there?

Reply:
I meant to say that the target created within the if-block depends upon ${DDS_ROOT}/tests/DCPS/Messenger. So, in order to make these run outside of the source tree the following would have to be copied:

* ${DDS_ROOT}/tests/DCPS/Messenger
* ${DDS_ROOT}/tests/cmake_integration/Messenger/Messenger_1
* Other perl dependencies to run things

Given that, the primary purpose is to test that the CMake build functions are working as expected (outside the source tree), not whether or not the perl scripts run them—for that they are being tested inside the source tree. It's not perfect, but it works to to the manual test!

Reply:
I would recommend putting a comment above the if-block just to note what I had to explain, but other than that I'm not sure it's worth the time spent to remove these; otherwise the make-install scenario needs to be tested manually and and the documentation will have to be changed/tested for not much benefit.

Reply:
Okay, but I don't think you answered my question: why does it have to copied outside the source tree at all? We're not doing that on the jenkins install tests, but is there something that the cmake module does differently inside the source tree vs outside it?

Reply:
Sorry Fred I didn't answer your previous question! I suppose it wouldn't have to be copied out of the source tree—although early on in development when the CMake package was a CMake module I think there was a clash of the DDS_ROOTs going on.

However, I think it still keeps with the separation logically—suppose I downloaded a pre-built OpenDDS but I grabbed the cmake_integration/Messenger/* tests to run the simple example, I would have to pull in all the other stuff too.

Reply:
> We're not doing that on the jenkins install tests, but is there something that the cmake module does differently inside the source tree vs outside it?

The CMake package doesn't do anything differently—it all depends on the CMAKE_PREFIX_PATH. But, as I stated before, copying this outside of the source root makes for the example in the documentation—which is a manual exercise to build outside of the source tree. This is separate from automated tests.

Reply:
> But, as I stated before, copying this outside of the source root makes for the example in the documentation—which is a manual exercise to build outside of the source tree. This is separate from automated tests.

This is a bit strange, building something that doesn't run. Perhaps a new example can be concocted? It would be nice not to have a dependency on the Perl testing libraries, so something more standalone would be ideal! Because really we're using a unit test in a way that it shouldn't really.